.github/workflows/build.yaml

```diff
@@ -66,13 +65,55 @@ jobs:
         with:
           path: ./website/public
 
+  api-docs:
+    uses: ./.github/workflows/api-docs.yaml
+    secrets: inherit
+
+  adr-index:
+    uses: ./.github/workflows/adr-index.yaml
+    secrets: inherit
+
+  # Build job
+  architecture:
+    needs:
+      - api-docs
+      - adr-index
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: tmp/architecture
+          repository: konflux-ci/architecture
+      - name: Download generated API docs
+        uses: actions/download-artifact@v3
+        with:
+          name: api-docs
+          path: tmp/architecture/ref/
+      - name: Download generated ADR index
+        uses: actions/download-artifact@v3
+        with:
+          name: adr-index
+          path: tmp/architecture/ADR/
+      - name: Setup Pages
+        uses: actions/configure-pages@v2
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./tmp/architecture/
+          destination: ./_site
```

Comment:
What's the end goal of this process?
I understand that we're generating API docs and gather some markdown files.

In this stage here we build with Jekyll and move the artifacts to `_site` dir, while on the workflow deployed at the moment we build with Hugo and move the artifacts from the `website/public` directory.

It seems to me that we're using two different deployments mechanisms from different locations to deploy to the same github pages site.

How are those new pages are going to be reached?

Reply:
The way I see it, we should generate a single artifact that includes both the current content and the architecture content. Hugo should be capable of generating HTMLs based on the rendered markdown files.
We should then change the landing page to link to the architecture index.

The theme that we're using should give us a few ways to create that linkage. See some examples of what can be done with the theme here: https://devcows.github.io/hugo-universal-theme/

Reply:
Yes, that's good feedback. I'll adopt it!

Reply:
I made a little progress today. I'm at a point now where I need to decide how much we are or aren't going to restructure the architecture repo to match this.

I guess there's no value in maintaining it as a jekyll deployment. It might as well switch entirely to hugo as well.